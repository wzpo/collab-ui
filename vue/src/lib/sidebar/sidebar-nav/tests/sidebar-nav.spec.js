import { shallowMount } from '@vue/test-utils';
import SidebarNav from '../index.vue';
import List from '../../../list/index.vue';
import Vue from 'vue';

Vue.component(List.name, List);

describe('SidebarNav', () => {
  it('should match snapshot', () => {
    const wrapper = shallowMount(SidebarNav, {
      propsData: {
        navSectionTitle: 'Overview'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should add customized class name if className prop is set', () => {
    const wrapper = shallowMount(SidebarNav, {
      propsData: {
        className: 'testClassName'
      }
    });

    expect(wrapper.find('.testClassName').exists()).toBeTruthy();
  });

  it('should render children if children prop is set', () => {
    const wrapper = shallowMount(SidebarNav, {
      slots: {
        default: `<div class='dummy-children'>Dummy Children</div>`
      }
    });

    expect(wrapper.find('.dummy-children').exists()).toBeTruthy();
  });


  it('should render header if title set', () => {
    const wrapper = shallowMount(SidebarNav, {
      propsData: {
        title: 'testClassName'
      }
    });

    expect(wrapper.find('.md-sidebar-nav__header').exists()).toBeTruthy();
  });

  it('should render custom headerNode if passed in', () => {
    const wrapper = shallowMount(SidebarNav, {
      slots: {
        header: `<div class='dummy-header'>Dummy Children</div>`
      }
    });

    expect(wrapper.find('.dummy-header').exists()).toBeTruthy();
  });
});

