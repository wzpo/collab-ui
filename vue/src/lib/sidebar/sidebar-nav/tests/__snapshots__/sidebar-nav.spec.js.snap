// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarNav should match snapshot 1`] = `
<div class="md-sidebar-nav" navsectiontitle="Overview">
  <!---->
  <md-list-stub focusfirst="true" focusfirstquery="" focusquery="
      .md-sidebar-nav__group--primary > .md-list-item:not(.disabled):not(:disabled):not(.md-list-item--read-only),
      .md-sidebar-nav__group--secondary.md-sidebar-nav__group--expanded > .md-list-item:not(.disabled):not(:disabled):not(.md-list-item--read-only),
      .md-sidebar-nav__group--secondary.md-sidebar-nav__group--expanded > .md-sidebar-nav__group--expanded > .md-list-item:not(.disabled):not(:disabled):not(.md-list-item--read-only)
      
    " itemrole="listitem" role="list" shouldloop="true" tabtype="vertical" trackactive="true" navsectiontitle="Overview" class="md-sidebar-nav__group md-sidebar-nav__group--primary"></md-list-stub>
</div>
`;
